bin/database.py

```diff
@@ -1054,6 +1054,65 @@ def upgrade_to_29():
     users = config.db.users.find({})
     process_cursor(users, upgrade_to_29_closure)
 
+def upgrade_to_30_acquisitions_closure(acquisition):
+    permissions = acquisition.get('permissions', [])
+    for permission_ in permissions:
+        if permission_.get('site', False):
+            del permission_['site']
+
+    result = config.db.acquisitions.update_one({'_id': acquisition['_id']}, {'$set': {'permissions' : permissions}})
+    if result.modified_count == 0:
+        return "Failed to remove site field"
+    return True
+
+def upgrade_to_30_groups_closure(group):
+    # Roles neeeds to be changed to permissions eventually
+    permissions = group.get('roles', [])
+    for permission_ in permissions:
+        if permission_.get('site', False):
+            del permission_['site']
+
+    result = config.db.groups.update_one({'_id': group['_id']}, {'$set': {'roles' : permissions}})
+    if result.modified_count == 0:
+        return "Failed to remove site field"
+    return True
+
+def upgrade_to_30_projects_closure(project):
+    permissions = project.get('permissions', [])
+    for permission_ in permissions:
+        if permission_.get('site', False):
+            del permission_['site']
+
+    result = config.db.projects.update_one({'_id': project['_id']}, {'$set': {'permissions' : permissions}})
+    if result.modified_count == 0:
+        return "Failed to remove site field"
+    return True
+
+def upgrade_to_30_sessions_closure(session):
+    permissions = session.get('permissions', [])
+    for permission_ in permissions:
+        if permission_.get('site', False):
+            del permission_['site']
+
+    result = config.db.sessions.update_one({'_id': session['_id']}, {'$set': {'permissions' : permissions}})
+    if result.modified_count == 0:
+        return "Failed to remove site field"
+    return True
+
+
+def upgrade_to_30():
+
+    cursor = config.db.acquisitions.find({'permissions.site' : {'$exists' : True}})
+    process_cursor(cursor, upgrade_to_30_acquisitions_closure)
+
+    cursor = config.db.groups.find({'roles.site': {'$exists': True}})
+    process_cursor(cursor, upgrade_to_30_groups_closure)
+
+    cursor = config.db.projects.find({'permissions.site' : {'$exists' : True}})
+    process_cursor(cursor, upgrade_to_30_projects_closure)
+
+    cursor = config.db.sessions.find({'permissions.site': {'$exists' : True}})
+    process_cursor(cursor, upgrade_to_30_sessions_closure)
```

Comment:
We should definitely rebase this branch off master when the timestamp fix lands and maybe bring in the roles->permissions rename changes as well.